packages/ui/src/components/form-input/components/form-text-input.tsx

```diff
@@ -0,0 +1,51 @@
+import { forwardRef, useCallback, useRef } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
+
+import { TextInput, type TextInputProps } from '@components/inputs'
+
+const FormTextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
+  // Only access hook if it's available in the component tree
+  const formContext = useFormContext()
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const setRefs = (element: HTMLInputElement | null) => {
+    // Save to local ref
+    inputRef.current = element
+
+    // Forward to external ref
+    if (typeof ref === 'function') {
+      ref(element)
+    } else if (ref) {
+      ref.current = element
+    }
+  }
+
+  if (!formContext) {
+    throw new Error(
```

Comment:
If `FormInput.Text` is not using within the scope of `FormProvider` from `react-hook-from`, then it will throw an exception with proper message.